kernel/src/hil/symmetric_encryption.rs

```diff
@@ -4,38 +4,48 @@
 
 //! Interface for symmetric-cipher encryption
 //!
-//! see boards/imix/src/aes_test.rs for example usage
+//! (TODO) Update usage example.
 
-use crate::ErrorCode;
+use crate::{utilities::leasable_buffer::SubSliceMut, ErrorCode};
 
 /// Implement this trait and use `set_client()` in order to receive callbacks from an `AES128`
-/// instance.
+/// instance. This returns the provided source. The `dest` contains the result of the operation
+/// and in both cases the `SubSliceMut` provided to `crypt()` will be returned.
 pub trait Client<'a> {
-    fn crypt_done(&'a self, source: Option<&'static mut [u8]>, dest: &'static mut [u8]);
+    fn crypt_done(
+        &'a self,
+        source: Option<SubSliceMut<'static, u8>>,
+        dest: Result<SubSliceMut<'static, u8>, (ErrorCode, SubSliceMut<'static, u8>)>,
+    );
 }
 
 /// The number of bytes used for AES block operations.  Keys and IVs must have this length,
 /// and encryption/decryption inputs must be have a multiple of this length.
 pub const AES128_BLOCK_SIZE: usize = 16;
 pub const AES128_KEY_SIZE: usize = 16;
 
+/// Implement this trait for hardware supported AES128 operation.
 pub trait AES128<'a> {
-    /// Enable the AES hardware.
-    /// Must be called before any other methods
-    fn enable(&self);
+    /// Convience associated type for configuring the AES128 peripheral.
+    /// This is intended to allows users the ability to pass any
+    /// relevant information to configure the given hardware.
+    type T;
 
-    /// Disable the AES hardware
+    // (todo) likely want to remove enable / disable per discussion on
+    // https://github.com/tock/tock/pull/4500
+    fn enable(&self);
     fn disable(&self);
```

Comment:
We should clarify when secrets are deleted (I'm assuming when `disable()` is called), but it's worth being really clear